__test__/creds/VerifiableCredential.test.js

```diff
@@ -818,4 +822,37 @@ describe('Unit tests for Verifiable Credentials', () => {
     expect(properties).toContain('contact.email.domain.name');
     expect(properties).toContain('contact.email.domain.tld');
   });
+
+  it('Should generate each VC and test the empty filtering', async (done) => {
+    const validateSchemaJestStep = async (credentialDefinition) => {
+      const ucaArray = [];
+      credentialDefinition.depends.forEach((ucaDefinitionIdentifier) => {
+        const ucaDefinition = definitions.find(ucaDef => (
+          ucaDef.identifier === ucaDefinitionIdentifier
+        ));
+        const ucaJson = SchemaGenerator.buildSampleJson(ucaDefinition);
+        let value = ucaJson;
+        if (Object.keys(ucaJson).length === 1) {
+          [value] = Object.values(ucaJson);
+        }
+        const dependentUca = new Claim(ucaDefinition.identifier, value, ucaDefinition.version);
+        ucaArray.push(dependentUca);
+      });
+      const credential = new VC(credentialDefinition.identifier, `jest:test:${uuidv1()}`, null, ucaArray, 1);
+
+      await credential.requestAnchor();
+      await credential.updateAnchor();
+
+      const filteredCredential = credential.filter([]);
+      return Object.keys(filteredCredential.claim).length === 0;
+    };
+    const promises = [];
+    credentialDefinitions.forEach((credentialDefinition) => {
+      promises.push(validateSchemaJestStep(credentialDefinition));
+    });
+    Promise.all(promises).then((values) => {
+      values.forEach(isValid => expect(isValid).toBeTruthy());
+      done();
+    });
```

Comment:
Add a `.catch` to fail the test.

Reply:
Why would I put a catch if it can never throw any exception there?

Reply:
Because if throws any time then it's wrong and the test must fail with an explicit message.
Nowadays, if someone change the code and it throws a runtime error, then your test will fail by an uncaught promise...

Reply:
Runtime errors throws from the forEach in the first error possible, and burn the execution of the test, thus failing it, there is no point in catching any error here. If it fails for runtime exception, than it fails, catching would only hide the stacktrace

Reply:
Just tested now and by using this way will Jest will print the exception on console.
Looks good.